plugins/Installation/Controller.php

```diff
@@ -112,15 +111,11 @@ function systemCheck()
             __FUNCTION__
         );
 
-        $view->duringInstall = true;
+        /** @var DiagnosticService $diagnosticService */
+        $diagnosticService = StaticContainer::get('Piwik\Plugins\Diagnostics\DiagnosticService');
+        $view->diagnosticReport = $diagnosticService->runDiagnostics();
```

Comment:
I didn't do it because the service has a lot of dependencies (all the diagnostics) across many plugins, so I don't want to play with fire in a controller where some requests are updating Piwik's codebase (performances shouldn't really be a big deal, it's mostly stability). But yes in theory DI would be better.